File: app/src/main/java/com/huck/biblequiz/util/TextCleaner.kt
package com.huck.biblequiz.util

object TextCleaner {

    private val htmlTagRegex = Regex("<[^>]*>")

    fun clean(html: String): String {
        return html
            .replace(htmlTagRegex, "")
            .replace("&nbsp;", " ")
            .replace("&amp;", "&")
            .replace("&lt;", "<")
            .replace("&gt;", ">")
            .replace("&quot;", "\"")
            .replace("&#39;", "'")
            .trim()
    }
}

File: ios/BeSpecific/Utilities/TextCleaner.swift
import Foundation

enum TextCleaner {
    private static let htmlTagRegex = try! NSRegularExpression(pattern: "<[^>]*>")

    static func clean(_ html: String) -> String {
        let range = NSRange(html.startIndex..., in: html)
        var result = htmlTagRegex.stringByReplacingMatches(in: html, range: range, withTemplate: "")
        result = result.replacingOccurrences(of: "&nbsp;", with: " ")
        result = result.replacingOccurrences(of: "&amp;", with: "&")
        result = result.replacingOccurrences(of: "&lt;", with: "<")
        result = result.replacingOccurrences(of: "&gt;", with: ">")
        result = result.replacingOccurrences(of: "&quot;", with: "\"")
        result = result.replacingOccurrences(of: "&#39;", with: "'")
        return result.trimmingCharacters(in: .whitespaces)
    }
}
